devconf-outreachy/source/index.rst

```diff
@@ -0,0 +1,397 @@
+
+Aija's Internship @OpenStack
+----------------------------
+
+* How it all begun
+
+  - finding a project
+
+     - something new, but in familiar programming language
+     - mentors in similar time zone
+
+* The initial application:
+
+  - Challenges
+
+    - initial ticket
+    - transparency of other applying
+
+  - Successes
+
+    - dedicated tickets for application
+
+.. Things to talk about ^ (Aija)
+
+  my background: a full-stack web developer in enterprise information systems
+  using Java and C#. Used FOSS for personal needs.
+
+  Criteria to filter initial list. Try something new that haven't done before.
+  Best chances with projects in Python that have used in personal projects.
+
+  Similar time zone to avoid waiting for hours if get stuck with something.
+  If wouldn't be able to find the project using all criteria, this would be dropped.
+
+  OpenStack project matching all criteria.
+
+  Choice of the ticket was not successful - in the end it did not result in any code,
+  but only documentation updates.
+  Seemed that best tickets were already taken.
+
+  Visibility of other people applying and how they are going. New applicant appearing
+  after application deadline extended.
+
+  Despite the challenges of initial ticket,
+  dedicated tickets still a good starting point.
+
+OpenStack: everybody struggles
+------------------------------
+
+* Technology
+
+  - Code complexity and its sheer size is overwhelming
+  - The developer's workbench takes time to learn
+  - Not always possible to run OpenStack at home
+
+* Social
+
+  - Code reviews tend to be slow
+  - Sometimes feedback can look harsh
+  - Newcomers are prone to taking the feedback personally
+  - It takes time become a good community citizen
+
+.. Things to talk about ^ (dtantsur)
+
+Outreachy application period
+----------------------------
+
+In the eyes of a mentor:
+
+* Come up with isolated, yet non-trivial and useful tasks
+* On-board a handful of candidates to let them proceed
+* Pondering code reviews aiming to merge their work
+
+.. Things to talk about ^ (ietingof)
+
+  According to the program policies, during the application period
+  prospective interns should prove their worthiness by doing some
+  accountable work.
+
+  Once the application period is over, the mentors are to rank
+  all successful applicants and Outreachy organizers will work
+  on figuring out the budget.
+
+  Seasonal coding is not easy with OpenStack because... it's
+  OpenStack (see previous slide).
+
+  But we managed to come up with a bunch of isolated coding tasks
+  and let the candidates chose and assign them to themselves.
+
+Aija's survival hints
+---------------------
+
+* Code reviews
+
+  - Don't take it personally (really don't)
+  - But take responsibility for the code you write
+  - Essential part of programming
+  - Opportunity to learn
+
+* Asking questions
+
+* Working from home/remotely
```

Comment:
Working from home is a whole interesting topic, to which I personally can also contribute :) Maybe split into a slide of its own?

Reply:
It's probably worth a whole separate talk :) At the moment it is not clear if there is enough time for another slide. After initial measurements my slides already took so big proportion out of allocated time, that I shortened them and took some other points out. If something is put back, then I would like to put back some other challenges to have some variety. Could tell more after dry-run.

Reply:
We should plan a separate talk! May be at some drones club... ;)